ingest-v2/src/main/kotlin/com/microsoft/azure/kusto/ingest/v2/ManagedStreamingIngestClient.kt

```diff
@@ -0,0 +1,633 @@
+                                            database,
+                                            table = table,
+                                            data = data,
+                                            format =
+                                            props
+                                                .format,
+                                            ingestProperties =
+                                            props,
+                                        )
+                                    IngestResponse(
+                                        ingestionOperationId =
+                                        source.sourceId
+                                            .toString(),
+                                    )
+                                }
+
+                                else ->
+                                    throw IngestException(
+                                        "Unsupported source type for streaming: ${source::class.simpleName}",
+                                        isPermanent = true,
+                                    )
+                            }
+
+                        val duration =
+                            Duration.ofMillis(
+                                Instant.now(Clock.systemUTC()).toEpochMilli() -
+                                    attemptStartTime,
+                            )
+                        managedStreamingPolicy.streamingSuccessCallback(
+                            source,
+                            database,
+                            table,
+                            props,
+                            ManagedStreamingRequestSuccessDetails(
+                                duration,
+                            ),
+                        )
+
+                        logger.info(
+                            "Streaming ingestion succeeded for ${source::class.simpleName} on attempt $attempt. Duration: ${duration.toMillis()}ms",
+                        )
+                        response
+                    } catch (e: Exception) {
+                        logger.warn(
+                            "Streaming ingestion attempt $attempt failed: ${e.message}",
+                        )
+                        throw e
+                    }
+                },
+                onRetry = { attempt, ex, _ ->
+                    logger.debug(
+                        "Retrying streaming ingestion after attempt $attempt due to: ${ex.message}",
+                    )
+                    if (source is LocalSource) {
+                        try {
+                            source.reset()
+                        } catch (e: Exception) {
+                            logger.warn(
+                                "Failed to reset source stream: ${e.message}",
+                            )
+                        }
+                    }
+                },
+                shouldRetry = { attempt, ex, isPermanent ->
```

Comment:
We have to have a condition where maximum retries exceed. How it happens is not very clear in this block. We do not seem to have a counter yet for it